1.0.53 is broken on stable (#![feature]` may not be used on the stable release channel)

To reproduce
```sh
> rustc -vV
rustc 1.75.0 (82e1608df 2023-12-21)
binary: rustc
commit-hash: 82e1608dfa6e0b5569232559e3d385fea5a93112
commit-date: 2023-12-21
host: aarch64-apple-darwin
release: 1.75.0
LLVM version: 17.0.6
> cargo new --lib thiserr-test
> cargo add thiserror
# make sure version is 1.0.53
> cargo build
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /Users/muradb/.cargo/registry/src/index.crates.io-6f17d22bba15001f/thiserror-1.0.53/src/lib.rs:238:50
    |
238 | #![cfg_attr(error_generic_member_access, feature(error_generic_member_access))]
    |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Change version to `=1.0.52`.
Now `cargo build` works fine.

I checked that the CI for 1.0.53 has ran fine yesterday, so this might be an issue with OS (unlikely tho)?

--- a/build/probe.rs
+++ b/build/probe.rs
@@ -27,3 +27,6 @@ impl Error for MyError {
         request.provide_ref(&self.0);
     }
 }
+
+// Include in sccache cache key.
+const _: Option<&str> = option_env!("RUSTC_BOOTSTRAP");

